### PR TITLE
feat: Added velero installation and backup module

### PR DESCRIPTION
**Pull request by mamari90**
### Type of changes

- [x] Add new module
- [x] Update existing module
- [ ] Remove existing module

### Does this introduce a breaking change?

- [ ] Yes
- [x] No

### Run checks

Useful commands to run checks on local machine

```sh
bash .utils/terraform_run_all.sh init local
pre-commit run -a
```

### CODE REVIEW

File: kubernetes_cluster_velero/01_main.tf [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  name                  = var.backup_storage_container_name
+  storage_account_name  = data.azurerm_storage_account.velero_storage_account.name
+  container_access_type = "private"
+
+}
+
+data "azuread_client_config" "current" {}
+
+resource "azuread_application" "velero_application" {
+  display_name = "velero-application"
```
[the remark this answers is not in the document]
Comment (mamari90) [reply kubernetes_cluster_velero/01_main.tf:16]: done

File: kubernetes_cluster_velero/01_main.tf [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  provisioner "local-exec" {
+    when    = destroy
+    command = <<EOT
+    velero uninstall --force
+    EOT
+  }
+
+  provisioner "local-exec" {
+    command = <<EOT
+    velero install --provider azure --plugins velero/velero-plugin-for-microsoft-azure:v1.5.0 \
```
[the remark this answers is not in the document]
Comment (mamari90) [reply kubernetes_cluster_velero/01_main.tf:83]: done

File: kubernetes_cluster_velero/01_main.tf [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    when    = destroy
+    command = <<EOT
+    velero schedule delete ${self.triggers.name} --confirm
+    EOT
+  }
+
+  provisioner "local-exec" {
+    command = <<EOT
+    velero create schedule ${local.all_namespace_backup_name} --schedule="${var.backup_schedule}" --ttl ${var.backup_ttl} --snapshot-volumes ${var.volume_snapshot}
+    EOT
```
[the remark this answers is not in the document]
Comment (mamari90) [reply kubernetes_cluster_velero/01_main.tf:124]: not necessary, velero outputs this:
Schedule "prova" created successfully.

File: kubernetes_velero_backup/01_main.tf [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    when    = destroy
+    command = <<EOT
+    velero schedule delete ${self.triggers.backup_name}-${self.triggers.namespace} --confirm
+    EOT
+  }
+
+  provisioner "local-exec" {
+    command = <<EOT
+    velero schedule create ${self.triggers.backup_name}-${self.triggers.namespace} --schedule="${var.schedule}" --ttl ${var.ttl} --include-namespaces ${each.value}
+    EOT
```
[the remark this answers is not in the document]
Comment (mamari90) [reply kubernetes_velero_backup/01_main.tf:22]: not necessary, velero outputs this:
Schedule "prova" created successfully.

File: kubernetes_cluster_velero/01_main.tf [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+  resource_group_name = var.aks_cluster_rg
+}
+
+module "velero_storage_account" {
+  source = "git::https://github.com/pagopa/terraform-azurerm-v3.git//storage_account?ref=v7.2.0"
+
+  name                            = "${local.sa_prefix}velerosa"
+  account_kind                    = "BlobStorage"
+  account_tier                    = "Standard"
+  account_replication_type        = "LRS"
```
Comment (diegolagospagopa) [inline kubernetes_cluster_velero/01_main.tf:16]: please allow to choose either ZRS

**Review by diegolagospagopa — CHANGES_REQUESTED** [review]

File: kubernetes_cluster_velero/01_main.tf (hunk 5 above)
Comment (mamari90) [reply kubernetes_cluster_velero/01_main.tf]: done

**Review by diegolagospagopa — APPROVED** [review]

**Pull request merged**